### PR TITLE
Storage AutoYaST backward compatibility

### PR DESCRIPTION
**Pull request by ancorgs**
Created to replace #1266 (to sanitize history).

Agama is going to offer new settings for configuring the storage devices, see https://github.com/openSUSE/agama/pull/1256. That new settings will be applied by loading a config file (e.g., `$ agama config load myconfig.json`). Nevertheless, Agama is also going to temporary support the SLE15 AutoYaST schema to some extent, in order to make the transition smoother. For that, the json config file can indicate a "legacyAutoyastStorage" key equivalent to the partitioning section of the AutoYaST profile:

~~~
"legacyAutoyastStorage": [
  {
    "device": "/dev/vda",
    "use": "all",
    "partitions": [...]
  }
]
~~~ 

This PR implements the support for the "legacyAutoyastStorage" settings, making possible to calculate an AutoYaST proposal.

Converting the "partitioning" section from a XML AutoYaST profile to a "legacyAutoyastStorage" section is implemented at https://github.com/openSUSE/agama/pull/1285

Note:  The AutoYaST guided partitioning will not be supported, see https://doc.opensuse.org/projects/autoyast/#CreateProfile-Guided-Partitioning.

### CODE REVIEW

File: rust/agama-lib/src/store.rs [hunk 1]
```diff
@@ -35,11 +37,16 @@ impl<'a> Store<'a> {
...
+            storage: StorageStore::new(connection.clone()).await?,
+            storage_autoyast: StorageAutoyastStore::new(connection).await?,
         })
     }
 
-    /// Loads the installation settings from the D-Bus service
+    /// Loads the installation settings from the D-Bus service.
+    ///
+    /// NOTE: The storage AutoYaST settings cannot be loaded because they cannot be modified. The
+    /// ability of using the storage AutoYaST settings from a json config file is temporary and it
```
Comment (imobachgs) [inline rust/agama-lib/src/store.rs:48]:
```suggestion
    /// ability of using the storage AutoYaST settings from a JSON config file is temporary and it
```

File: service/lib/agama/storage/proposal_strategies/guided.rb [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+          device = available_devices.first&.name
+          return unless device
+
+          case settings.device
+          when DeviceSettings::Disk
+            settings.device.name = device
+          when DeviceSettings::NewLvmVg
+            settings.device.candidate_pv_devices = [device]
+          when DeviceSettings::ReusedLvmVg
+            # TODO: select an existing VG?
```
Comment (imobachgs) [inline service/lib/agama/storage/proposal_strategies/guided.rb:87]: Is this something to fix now?

**Review by imobachgs — APPROVED** [review]
LGTM

File: service/lib/agama/storage/proposal_strategies/guided.rb (hunk 2 above)
Comment (ancorgs) [reply service/lib/agama/storage/proposal_strategies/guided.rb:87]: No. We have not yet implemented reusing LVMs.

File: rust/agama-derive/src/lib.rs [hunk 3]
```diff
@@ -176,7 +177,7 @@ fn expand_merge_fn(settings: &SettingFieldsList) -> TokenStream2 {
     let arms = settings.all().iter().map(|s| {
         let field_name = &s.ident;
         match s.kind {
-            SettingKind::Scalar => quote! {
+            SettingKind::Scalar | SettingKind::Ignored => quote! {
```
[the remark this answers is not in the document]
Comment (ancorgs) [reply rust/agama-derive/src/lib.rs:180]: Ces't la vie.

**Review by imobachgs — APPROVED** [review]

**Pull request merged**